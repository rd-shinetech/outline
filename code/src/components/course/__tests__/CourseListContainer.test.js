import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import createMemoryHistory from 'history/createMemoryHistory';
import { CourseListContainer } from '../CourseListContainer';


describe('CourseListContainer.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            courses: [
                { title: 'Java The Good Pards' }
            ],
            action: { getCoursesAction: jest.fn() },
            history: createMemoryHistory()
        };

        const wrapper = shallow(<CourseListContainer {...props} />);

        expect(wrapper).toHaveLength(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();

        expect(wrapper.find('button')).toHaveLength(3);
    });

  
});

