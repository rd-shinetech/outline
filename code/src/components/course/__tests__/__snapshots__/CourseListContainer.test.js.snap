exports[`CourseListContainer.test.js renders without crashing 1`] = `
<div
  className="container-fluid">
  <div
    className="row mt-3">
    <div
      className="col">
      <h1>
        Courses
      </h1>
    </div>
  </div>
  <div
    className="row mt-3">
    <div
      className="col">
      <div
        className="btn-group"
        role="group">
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-plus" />
           New
        </button>
        <button
          className="btn btn-warning ml-2"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-pencil" />
           Edit
        </button>
        <button
          className="btn btn-danger ml-2"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-trash-o"
            onClick={[Function]} />
           Delete
        </button>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col">
      <CourseList
        courses={
          Array [
            Object {
              "title": "Java The Good Pards",
            },
          ]
        }
        handleRowSelect={[Function]} />
    </div>
  </div>
</div>
`;
